action_create.go

```diff
@@ -0,0 +1,130 @@
+package devicemngmt
+
+import (
+	"context"
+	"errors"
+
+	"github.com/Selly-Modules/logger"
+	"github.com/Selly-Modules/mongodb"
+	ua "github.com/mssola/user_agent"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+// DeviceCreate ...
+type DeviceCreate struct {
+	DeviceID   string
+	UserID     string
+	UserAgent  string
+	AppVersion string
+	IP         string
+	FCMToken   string
+	Language   string
+}
+
+// Create ...
+func (s Service) Create(payload DeviceCreate) error {
+	var (
+		col = s.getDeviceCollection()
+		ctx = context.Background()
+	)
+
+	// New device data from payload
+	deviceData, err := payload.newDevice()
+	if err != nil {
+		return err
+	}
+
+	// Find device id existed or not
+	device := Device{}
+	if err = col.FindOne(ctx, bson.M{"deviceID": deviceData.DeviceID}).Decode(&device); err != nil {
+		logger.Error("devicemngt - findByDeviceID", logger.LogData{
+			"deviceID": deviceData.DeviceID,
+			"err":      err.Error(),
+		})
+	}
+	if !device.ID.IsZero() {
+		return errors.New("deviceID already exists")
```

Comment:
deviceID already exists -> this device is already existed

Reply:
done